services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbDeviceRegistryUtils.java

```diff
@@ -139,4 +142,21 @@ private MongoDbDeviceRegistryUtils() {
                     String.format("Resource [%s] not found.", resourceId)));
         }
     }
+
+    /**
+     * Checks if a provided error is {@link MongoException} with error category {@link ErrorCategory#DUPLICATE_KEY}.
```

Comment:
How about explaining the purpose of the method in the description? Something like `Checks if the given error is caused due to duplicate keys.`

Reply:
done